packages/flutter/lib/src/widgets/page_storage.dart

```diff
@@ -129,7 +112,117 @@ class PageStorageBucket {
   }
 }
 
-/// A widget that establishes a page storage bucket for this widget subtree.
+/// Establish a subtree in which widgets can opt into persisting states after
+/// being destroyed.
+///
+/// [PageStorage] is used to save and restore values that can outlive the widget.
+/// For example, when multiple pages are grouped in tabs, when a page is
+/// switched out, its widget is destroyed and its state is lost. By adding a
+/// [PageStorage] at the root and adding a [PageStorageKey] to each page, most of the
+/// page's state will be stored in its closest ancestor [PageStorage], and
+/// restored when it's switched back. Only the state knowing how to use [PageStorage] is persisted.
+///
+/// Usually you don't need to explicitly use a [PageStorage], since it's already
+/// included in routes.
+///
+/// [PageStorageKey] is used by [Scrollable] if
+/// `keepScrollOffset` is enabled to save their [ScrollPosition]s.
+/// {@tool dartpad --template=freeform}
+///
+/// This sample shows how to explicitly use a [PageStorage] to
+/// store the states of its children pages. Each page includes a scrollable
+/// list, whose position is preserved when switching between the tabs thanks to
+/// the help of [PageStorageKey].
+///
+/// ```dart imports
+/// import 'package:flutter/material.dart';
+/// ```
+///
+/// ```dart main
+/// void main() => runApp(MyApp());
```

Comment:
There appears to be some variability in the indentations and such. Can you check the style guide and make sure the code is formatted? :)